server/src/main/java/org/opensearch/index/mapper/DerivedFieldMapper.java

```diff
@@ -38,62 +44,185 @@ private static DerivedFieldMapper toType(FieldMapper in) {
      * @opensearch.internal
      */
     public static class Builder extends ParametrizedFieldMapper.Builder {
-        // TODO: The type of parameter may change here if the actual underlying FieldType object is needed
-        private final Parameter<String> type = Parameter.stringParam("type", false, m -> toType(m).type, "");
+        private final Parameter<String> type = Parameter.stringParam("type", true, m -> toType(m).type, "");
+        private final IndexAnalyzers indexAnalyzers;
+        private final boolean defaultIgnoreMalformed;
+        private final DateFormatter defaultDateFormatter;
 
         private final Parameter<Script> script = new Parameter<>(
             "script",
-            false,
+            true,
             () -> null,
             (n, c, o) -> o == null ? null : Script.parse(o),
             m -> toType(m).script
         ).setSerializerCheck((id, ic, value) -> value != null);
 
-        public Builder(String name) {
+        private final Parameter<Map<String, Object>> properties = new Parameter<>(
+            "properties",
+            true,
+            Collections::emptyMap,
+            (n, c, o) -> parseProperties(n, o),
+            m -> toType(m).properties
+        );
+        private final Parameter<String> sourceIndexedField = Parameter.stringParam(
+            "source_indexed_field",
```

Comment:
yeah, that sounds better :)

Reply:
No, its the text field, an optional field, which can be associated with object field to filter results in 2 phased iterator approach to make query faster. 
Do you think I should rename it?

Reply:
Oh -- got it.

Maybe `prefilter_field`? 

Alternatively, we could use `MatchAllDocs` as the first phase (with `maxDoc` or `Long.MAX_VALUE` as the `cost()`) for the object query, then put all the logic in the second phase. The query on the text field could be a simple query in a conjunction. That is, you could do:

```
"query": {
  "bool": {
    "must": [{
        "term": {
          "derived_object.last_name": "smith"
        }
      }],
    "filter": [{
        "term": {
          "bag_of_words": "smith"
        }
      }]
    }
  }
}
```

In this case, the filter on `bag_of_words` would have cost equal to the `docFreq` of that term, so it's guaranteed to be the lead iterator. That is, we don't need it to be an attribute of the derived object field, I think.

What do you think?

Reply:
I liked this alternative as it will result in better cost estimation and query planning vs current approach. I have implemented it in current version, please take a look if that's what you mean.